extensions/parquet/table/src/main/java/io/deephaven/parquet/table/layout/ParquetFlatPartitionedLayout.java

```diff
@@ -63,8 +65,16 @@ public String toString() {
     public void findKeys(@NotNull final Consumer<ParquetTableLocationKey> locationKeyObserver) {
         final SeekableChannelsProvider provider = SeekableChannelsProviderLoader.getInstance().fromServiceLoader(
                 tableRootDirectory, readInstructions.getSpecialInstructions());
+        final boolean isFileURI = FILE_URI_SCHEME.equals(tableRootDirectory.getScheme());
         try (final Stream<URI> stream = provider.list(tableRootDirectory)) {
-            stream.filter(ParquetUtils::isVisibleParquetURI).forEach(uri -> {
+            stream.filter(uri -> {
+                if (isFileURI) {
```

Comment:
I might suggest pulling the bifurcation out of the inner loop; either, creating a `Predicate<URI>`, or having two different call sites to `.filter`.